salt/utils/xmlutil.py

```diff
@@ -379,3 +379,35 @@ def change_xml(doc, data, mapping):
                 deleted = del_fn(parent_map, node)
                 need_update = need_update or deleted
     return need_update
+
+
+def visit(node, fn):
+    """
+    Recursively apply a function on an XML node and its children.
+
+    :param node: the ElementTree node to apply the function on
+    :param fn: the function to apply. Takes the node as only parameter and no return is expected.
+    :return: the node
+    """
+    fn(node)
+    for child in node:
+        visit(child, fn)
+    return node
+
+
+def strip_spaces(node):
+    """
+    Remove all spaces and line breaks before and after nodes.
+    This helps comparing XML trees.
+
+    :param node: the XML node to remove blanks from
+    :return: the node
+    """
+
+    def space_stripper(node):
+        if node.tail is not None:
+            node.tail = node.tail.strip(" \t\n")
+        if node.text is not None:
+            node.text = node.text.strip(" \t\n")
```

Comment:
**suggestion**: I don't think that we gain anything by having this function as an internal function. We could either make this an internal function, e.g.

```
def _space_stripper(...):
```

Or, if we're going to use recursion we can just simplify this:

```
def strip_spaces(node):
    '''...'''

    if node.tail is not None:
        node.tail = node.tail.strip(" \t\n")
    if node.text is not None:
        node.text = node.text.strip(" \t\n")
    try:
        for child in node:
             strip_spaces(child)
    except RecursionError:
        raise Exception('but something better, OK?')

    return node
```

by removing the `visit` call altogether.

Reply:
OK, I'll go for the version without the `visit` call: looks cleaner